packages/eui/src/components/drag_and_drop/drag_drop_context.stories.tsx

```diff
@@ -64,3 +71,187 @@ export const Playground: Story = {
+    });
+
+    await setTimeout(async () => {
+      await waitFor(async () => {
+        await fireEvent.mouseDown(canvas.getByTestSubject('draggable-item-1'));
+        await fireEvent.mouseMove(canvas.getByTestSubject('draggable-item-1'), {
+          clientX: 0,
+          clientY: 5,
+        });
+
+        expect(
+          [...canvas.getByTestSubject('draggable-item-1').classList]
+            .join('')
+            .includes('isDragging')
+        ).toBe(true);
+      });
+    }, 150); // add a timeout to prevent differences due to animation
+  },
+};
+
+export const WithinModals: Story = {
+  tags: ['vrt-only'],
+  parameters: {
+    loki: {
+      skip: false,
+      chromeSelector: LOKI_SELECTORS.portal,
+    },
+  },
+  args: {
+    children: (
+      <EuiDroppable droppableId="droppableArea">
+        <EuiDraggable
+          spacing="m"
+          index={0}
+          draggableId="draggable-item-1"
+          usePortal
+          data-test-subj="draggable-item-1"
+        >
+          {(_, state) => (
+            <EuiPanel hasShadow={state.isDragging}>
+              Draggable item 1 {state.isDragging && '✨'}
+            </EuiPanel>
+          )}
+        </EuiDraggable>
+        <EuiDraggable
+          spacing="m"
+          index={1}
+          draggableId="draggable-item-2"
+          usePortal
+          data-test-subj="draggable-item-2"
+        >
+          {(_, state) => (
+            <EuiPanel hasShadow={state.isDragging}>
+              Draggable item 2 {state.isDragging && '✨'}
+            </EuiPanel>
+          )}
+        </EuiDraggable>
+      </EuiDroppable>
+    ),
+  },
+  render: (args) => <FlyoutOrModalStory type="modal" {...args} />,
+  play: async ({ canvasElement }: PlayFunctionContext<ReactRenderer>) => {
+    const canvas = within(canvasElement);
+
+    await waitFor(async () => {
+      expect(canvas.getByRole('dialog')).toBeInTheDocument();
+      expect(canvas.getByRole('dialog')).toBeVisible();
+    });
+
+    await setTimeout(async () => {
+      await waitFor(async () => {
+        await fireEvent.mouseDown(canvas.getByTestSubject('draggable-item-1'));
+        await fireEvent.mouseMove(canvas.getByTestSubject('draggable-item-1'), {
+          clientX: 0,
+          clientY: 5,
+        });
+
+        expect(
+          [...canvas.getByTestSubject('draggable-item-1').classList]
+            .join('')
+            .includes('isDragging')
+        ).toBe(true);
+      });
+    }, 150);
+  },
+};
+
+const FlyoutOrModalStory = ({
+  type,
+  ...args
+}: DragDropContextProps & { type: 'flyout' | 'modal' }) => {
+  const [isOpen, setOpen] = useState(true);
+
+  const handleOnClose = () => setOpen(false);
+
+  return (
+    <>
+      <EuiButton onClick={() => setOpen(!isOpen)}>Toggle</EuiButton>
+      {isOpen && (
+        <>
+          {type === 'flyout' ? (
+            <EuiFlyout
+              ownFocus
+              onClose={handleOnClose}
```

Comment:
see above, this can just be a noop
```suggestion
              onClose={() => {}}
```